packages/dialtone-vue2/components/icon/icon.vue

```diff
@@ -62,13 +61,25 @@ export default {
     },
   },
 
+  data () {
+    return {
+      icon: null,
+    };
+  },
+
   computed: {
     iconSize () {
       return ICON_SIZE_MODIFIERS[this.size];
     },
+  },
 
-    icon () {
-      return icons[`./${this.name}.vue`];
+  watch: {
+    name: {
+      immediate: true,
+      async handler (name) {
+        const icon = await import(`/node_modules/@dialpad/dialtone-icons/vue2/dist/components/${name}.js`);
+        this.icon = icon.default;
+      },
```

Comment:
I remember trying to do a dynamic import within our library before and it errored out when they tried to use it on the application side. Our infrastructure is much different than back then though, so maybe it works now. Just something to look out for when we try this out.

Reply:
Tested on multiple sample applications and working well so far so I don't expect any mayor issue while testing on product.

Reply:
k sounds good.